sys/vm/vm.h

```diff
@@ -79,10 +79,10 @@ typedef u_char vm_prot_t;	/* protection codes */
 #define	VM_PROT_READ		((vm_prot_t) 0x01)
 #define	VM_PROT_WRITE		((vm_prot_t) 0x02)
 #define	VM_PROT_EXECUTE		((vm_prot_t) 0x04)
-#define	VM_PROT_COPY		((vm_prot_t) 0x08)	/* copy-on-read */
-#define	VM_PROT_PRIV_FLAG	((vm_prot_t) 0x10)
-#define	VM_PROT_READ_CAP	((vm_prot_t) 0x20)
-#define	VM_PROT_WRITE_CAP	((vm_prot_t) 0x40)
+#define	VM_PROT_READ_CAP	((vm_prot_t) 0x08)
+#define	VM_PROT_WRITE_CAP	((vm_prot_t) 0x10)
+#define	VM_PROT_COPY		((vm_prot_t) 0x20)	/* copy-on-read */
+#define	VM_PROT_PRIV_FLAG	((vm_prot_t) 0x40)
```

Comment:
I wonder if it makes sense to move kernel-only protections to 0x10000 and up to align with the PROT_MAX space.

Reply:
So vm_prot_t is only a char, and LinuxKPI already shoves it into a uint16_t and uses the upper 8 bits.  However, in a followup I haven't pushed yet I moved these down to where PROT_READ_CAP/PROT_WRITE_CAP would be in case we add them since mmap/mprotect generally assume that PROT_xxx == VM_PROT_xxx.  If we wanted PROT_READ_CAP/PROT_WRITE_CAP we would have to have some sort of way of signaling that they are valid.  Still not sure about that.  PROT_NONE in the PROT_MAX mask worked ok, but there's not quite a simple way to do the same for these permissions.

Reply:
Ah, that makes sense. I didn't check vm_prot_t's type.